sbnci/PlottingScripts/sbnciplot_recoeff.C

```diff
@@ -0,0 +1,191 @@
+#include "TTree.h"
+#include "TFile.h"
+#include "TH1F.h"
+#include "TString.h"
+
+void EffPlot(TTree *tree, TFile* outputFile, TString directoryName, TString PDG, TString variable, 
+	     int nBins, float xMin, float xMax, TString title, bool goodReco = false);
+
+void Plot(TTree* tree, TFile* outputFile, TString directoryName, TString PDG, TString variable, 
+          int nBins, float xMin, float xMax, TString title, TString variableName);
+
+ void sbnciplot_recoeff(TString inputFileName)
+ {
+  TFile *inputFile = TFile::Open(inputFileName.Data());
+  TTree *tree = (TTree*) inputFile->Get("recoEff/ParticleTree");
+
+  TFile *outputFile = new TFile("ci_validation_histos.root","RECREATE");
+  gStyle->SetOptStat(0);
+  gROOT->ForceStyle();
+
+
+  outputFile->cd();
+  gDirectory->mkdir("muon");
+  gSystem->Exec(Form("mkdir -v %s","muon"));
+  gDirectory->mkdir("electron");
+  gSystem->Exec(Form("mkdir -v %s","electron"));
+  gDirectory->mkdir("proton");
+  gSystem->Exec(Form("mkdir -v %s","proton"));
+
+  //Muon Plots
+  EffPlot(tree,outputFile,"muon","13","mc_theta_xz",30,-180,180,";True #theta_{xz} (#circ);Fraction;");
+  EffPlot(tree,outputFile,"muon","13","mc_theta_yz",30,-180,180,";True #theta_{yz} (#circ);Fraction;");
+  EffPlot(tree,outputFile,"muon","13","mc_theta_xy",30,-180,180,";True #theta_{xy} (#circ);Fraction;");
```

Comment:
These hist definitions (and other below) tend to share input values.  It would reduce maintenance points if the values were defined in const vars above here.

Reply:
Yeah its a fair point, there is quite a lot of variation though, I think you'd end up defining so many options with different names that it wouldn't save much time for someone editing?